pkg/asset/manifests/openshift.go

```diff
@@ -159,6 +161,46 @@ func (o *Openshift) Generate(dependencies asset.Parents) error {
 			cloud.CACertFile = "/etc/kubernetes/static-pod-resources/configmaps/cloud-config/ca-bundle.pem"
 		}
 
+		// We also replace a username/password combination with an application credential
+		if cloud.AuthInfo != nil && (cloud.AuthInfo.UserID != "" || cloud.AuthInfo.Username != "") && cloud.AuthInfo.Password != "" {
+			conn, err := clientconfig.NewServiceClient("identity", opts)
+			if err != nil {
+				return err
+			}
+
+			// If the user has stored a user ID instead of username (uncommon but possible), use it.
+			userID := ""
+			if cloud.AuthInfo.UserID != "" {
+				userID = cloud.AuthInfo.UserID
+			} else {
+				authResult := conn.ProviderClient.GetAuthResult()
+				// We only support Identity API v3. v2 is waaaay old.
+				switch authResult := authResult.(type) {
+				case tokens.CreateResult:
+					u, err := authResult.ExtractUser()
+					if err != nil {
+						return err
+					}
+					userID = u.ID
+				default:
+					return errors.Errorf("Unsupported Identity API version")
+				}
+			}
+
+			// Gimme gimme gimme 🎶
+			createOpts := applicationcredentials.CreateOpts{
+				// TODO: Generate/retrieve a unique, per-install name
+				Name: "openshift-installer",
+			}
+			applicationCredential, err := applicationcredentials.Create(conn, userID, createOpts).Extract()
+
+			cloud.AuthInfo.Username = ""
+			cloud.AuthInfo.UserID = ""
+			cloud.AuthInfo.Password = ""
```

Comment:
I believe you also have to remove project (or tenant) name, id and domain name, user domain name and set `auth_type` to `v3applicationcredential`

Reply:
Done. I just recreate it now.